### ISSUE TITLE
add t distribution and robust EP 

### ISSUE THREAD

**Issue by yorkerlin** [shogun-toolbox/shogun#3450]
@karlnapf 
It will be great to add t distribution and robust EP for GP regression.
Note that the existing t distribution does not support EP inference. 

reference
paper: http://www.jmlr.org/papers/volume12/jylanki11a/jylanki11a.pdf
code:  http://research.cs.aalto.fi/pml/software/gpstuff/

**Comment by karlnapf** [comment]
Yep, this also depends on finally tackling a proper base class for distributions, #1998, #1929

**Comment by yorkerlin** [comment]
I took a close look at the implementation in gpstuff. 
Note that the EP implementation in shogun does not pay much attention to negative covariance.

**Comment by yorkerlin** [comment]
this issue is related to issue https://github.com/shogun-toolbox/shogun/issues/1832

**Comment by yorkerlin** [comment]
reference:
https://arxiv.org/abs/1412.4869
